python/doc/examples/data_analysis/distribution_fitting/plot_advanced_mle_estimator.py

```diff
@@ -0,0 +1,289 @@
+#
+# - :math:`\mu \in [-10, +10]` ;
+# - :math:`\sigma \in [\epsilon, +10]` ;
+# - :math:`a \in [-1 - \epsilon, \bar{x} - \epsilon]` ;
+# - :math:`b \in [\bar{x} + \epsilon, 1 + \epsilon]`
+#
+# where :math:`\bar{x}` is the sample mean and :math:`\epsilon`
+# is a small parameter.
+
+# %%
+epsilon = 1.0e-3
+bounds_lower = [-10.0, epsilon, -1.0 - epsilon, sample_mean + epsilon]
+bounds_upper = [10.0, 10.0, sample_mean - epsilon, 1.0 + epsilon]
+print("bounds_lower=", bounds_lower)
+print("bounds_upper=", bounds_upper)
+
+# %%
+# The boolean `False` means unbounded and `True` means bounded.
+
+# %%
+finiteLowerBound = [True, True, True, True]
+finiteUpperBound = [True, True, True, True]
+interval = ot.Interval(bounds_lower, bounds_upper, finiteLowerBound, finiteUpperBound)
+
+# %%
+# Set the solver and bounds
+# -------------------------
+
+# %%
+# The two methods that we suggest are based on the :class:`~openturns.MaximumLikelihoodFactory` class
+# which uses the maximum likelihood estimator to estimate
+# the parameters of the distribution depending on the sample.
+# The first method is to use the :meth:`~openturns.MaximumLikelihoodFactory.setOptimizationAlgorithm`
+# method.
+# This sets the optimization algorithm as a replacement for the
+# default optimization solver.
+
+
+def printTruncatedNormalParameters(truncatedNormalDistribution):
+    mu, sigma, a, b = truncatedNormalDistribution.getParameter()
+    print(" ", truncatedNormalDistribution.getImplementation().getClassName())
+    print("  mu= %.3f" % (mu))
+    print("  sigma= %.4f" % (sigma))
+    print("  a= %.2f" % (a))
+    print("  b= %.2f" % (b))
+    return None
+
+
+# %%
+factory = ot.MaximumLikelihoodFactory(distribution)
+factory.setOptimizationBounds(interval)
+distribution_MLE = factory.build(sample)
+print("Fitted distribution with default solver:")
+printTruncatedNormalParameters(distribution_MLE)
+default_solver = factory.getOptimizationAlgorithm()
+print("Default solver=", default_solver.getImplementation().getClassName())
+solver = ot.NLopt("LN_COBYLA")
+print("New solver=", solver.getClassName(), solver.getAlgorithmName())
+solver.setStartingPoint(startingPoint)
+factory.setOptimizationAlgorithm(solver)
+print(
+    "Updated solver=",
+    factory.getOptimizationAlgorithm().getImplementation().getClassName(),
+)
+distribution_MLE = factory.build(sample)
+print("Fitted distribution with new solver:")
+printTruncatedNormalParameters(distribution_MLE)
+
+
+# %%
+# Use ResourceMap to set the solver
+# ---------------------------------
+
+# %%
+# Another method is to use a special key of the :class:`ResourceMap`,
+# which defines the default solver used by :class:`~openturns.MaximumLikelihoodFactory`.
+# In this case, we do not define the starting point.
+
+# %%
+ot.ResourceMap.SetAsString(
+    "MaximumLikelihoodFactory-DefaultOptimizationAlgorithm", "LN_COBYLA"
+)
+factory = ot.MaximumLikelihoodFactory(distribution)
+factory.setOptimizationBounds(interval)
+print(
```

Comment:
This only returns 'Default solver= NLopt'

Reply:
This is because "LN_COBYLA" is one solver from NLopt. I think the text is unclear on this particular point. What may I write to clarify it?